.reuse/dep5

```diff
@@ -0,0 +1,8 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: cloud-hypervisor
+Upstream-Contact:  <>
+Source: https://www.cloudhypervisor.org
+
+Files: scripts/* test_data/* *.toml *.md .git* fuzz/Cargo.lock fuzz/.gitignore resources/linux-config-* vmm/src/api/openapi/cloud-hypervisor.yaml CODEOWNERS Cargo.lock
```

Comment:
Could you explain what this is?

Reply:
This is to bulk-license files. Without this reuse fails with:

```
# MISSING COPYRIGHT AND LICENSING INFORMATION

The following files have no copyright and licensing information:
* .github/ISSUE_TEMPLATE/bug_report.md
* .github/dependabot.yml
* .github/workflows/audit.yaml
* .github/workflows/build.yaml
* .github/workflows/dco.yaml
* .github/workflows/docker-image.yaml
* .github/workflows/fuzz-build.yaml
* .github/workflows/gitlint.yaml
* .github/workflows/hadolint.yaml
* .github/workflows/integration-arm64.yaml
* .github/workflows/integration-metrics.yaml
* .github/workflows/integration-rate-limiter.yaml
* .github/workflows/integration-sgx.yaml
* .github/workflows/integration-vfio.yaml
* .github/workflows/integration-windows.yaml
* .github/workflows/integration-x86-64.yaml
* .github/workflows/openapi.yaml
* .github/workflows/quality.yaml
* .github/workflows/release.yaml
* .github/workflows/reuse.yaml
* .github/workflows/shlint.yaml
* .gitignore
* .gitlint
* .rustfmt.toml
* .typos.toml
* CODEOWNERS
* CODE_OF_CONDUCT.md
* CONTRIBUTING.md
* CREDITS.md
* Cargo.lock
* Cargo.toml
* MAINTAINERS.md
* README.md
* api_client/Cargo.toml
* arch/Cargo.toml
* block/Cargo.toml
* devices/Cargo.toml
* docs/amd_sev_snp.md
* docs/api.md
* docs/balloon.md
* docs/building.md
* docs/coverage.md
* docs/cpu.md
* docs/custom-image.md
* docs/debug-port.md
* docs/device_model.md
* docs/fs.md
* docs/fuzzing.md
* docs/gdb.md
* docs/heap-profiling.md
* docs/hotplug.md
* docs/intel_sgx.md
* docs/intel_tdx.md
* docs/io_throttling.md
* docs/iommu.md
* docs/live_migration.md
* docs/logging.md
* docs/macvtap-bridge.md
* docs/memory.md
* docs/mshv.md
* docs/performance_metrics.md
* docs/profiling.md
* docs/releases.md
* docs/seccomp.md
* docs/snapshot_restore.md
* docs/tpm.md
* docs/tracing.md
* docs/uefi.md
* docs/vdpa.md
* docs/vfio-user.md
* docs/vfio.md
* docs/vhost-user-blk-testing.md
* docs/vhost-user-net-testing.md
* docs/virtiofs-root.md
* docs/vsock.md
* docs/windows.md
* event_monitor/Cargo.toml
* fuzz/.gitignore
* fuzz/Cargo.lock
* fuzz/Cargo.toml
* hypervisor/Cargo.toml
* net_gen/Cargo.toml
* net_util/Cargo.toml
* option_parser/Cargo.toml
* pci/Cargo.toml
* performance-metrics/Cargo.toml
* rate_limiter/Cargo.toml
* release-notes.md
* resources/linux-config-aarch64
* resources/linux-config-x86_64
* scripts/check-image-compatibility.sh
* scripts/common-aarch64.sh
* scripts/create-cloud-init.sh
* scripts/gitlint/rules.py
* scripts/prepare_vdpa.sh
* scripts/run_integration_tests_aarch64.sh
* scripts/run_integration_tests_live_migration.sh
* scripts/run_integration_tests_rate_limiter.sh
* scripts/run_integration_tests_sgx.sh
* scripts/run_integration_tests_vfio.sh
* scripts/run_integration_tests_windows_aarch64.sh
* scripts/run_integration_tests_windows_x86_64.sh
* scripts/run_integration_tests_x86_64.sh
* scripts/run_metrics.sh
* scripts/run_openapi_tests.sh
* scripts/run_unit_tests.sh
* scripts/sha1sums-aarch64
* scripts/sha1sums-x86_64
* scripts/test-util.sh
* scripts/windows-11-iot-enterprise-aarch64.raw.sha1
* serial_buffer/Cargo.toml
* test_data/cloud-init/ubuntu/ci/meta-data
* test_data/cloud-init/ubuntu/ci/network-config
* test_data/cloud-init/ubuntu/ci/user-data
* test_data/cloud-init/ubuntu/local/meta-data
* test_data/cloud-init/ubuntu/local/network-config
* test_data/cloud-init/ubuntu/local/user-data
* test_infra/Cargo.toml
* tpm/Cargo.toml
* tracer/Cargo.toml
* vhost_user_block/Cargo.toml
* vhost_user_net/Cargo.toml
* virtio-devices/Cargo.toml
* vm-allocator/Cargo.toml
* vm-device/Cargo.toml
* vm-migration/Cargo.toml
* vm-virtio/Cargo.toml
* vmm/Cargo.toml
* vmm/src/api/openapi/cloud-hypervisor.yaml

```

Reply:
> This is to bulk-license files. Without this reuse fails with:
> 
> ```
> # MISSING COPYRIGHT AND LICENSING INFORMATION
> 
> The following files have no copyright and licensing information:
> * .github/ISSUE_TEMPLATE/bug_report.md
> * .github/dependabot.yml
> * .github/workflows/audit.yaml
> * .github/workflows/build.yaml
> * .github/workflows/dco.yaml
> * .github/workflows/docker-image.yaml
> * .github/workflows/fuzz-build.yaml
> * .github/workflows/gitlint.yaml
> * .github/workflows/hadolint.yaml
> * .github/workflows/integration-arm64.yaml
> * .github/workflows/integration-metrics.yaml
> * .github/workflows/integration-rate-limiter.yaml
> * .github/workflows/integration-sgx.yaml
> * .github/workflows/integration-vfio.yaml
> * .github/workflows/integration-windows.yaml
> * .github/workflows/integration-x86-64.yaml
> * .github/workflows/openapi.yaml
> * .github/workflows/quality.yaml
> * .github/workflows/release.yaml
> * .github/workflows/reuse.yaml
> * .github/workflows/shlint.yaml
> * .gitignore
> * .gitlint
> * .rustfmt.toml
> * .typos.toml
> * CODEOWNERS
> * CODE_OF_CONDUCT.md
> * CONTRIBUTING.md
> * CREDITS.md
> * Cargo.lock
> * Cargo.toml
> * MAINTAINERS.md
> * README.md
> * api_client/Cargo.toml
> * arch/Cargo.toml
> * block/Cargo.toml
> * devices/Cargo.toml
> * docs/amd_sev_snp.md
> * docs/api.md
> * docs/balloon.md
> * docs/building.md
> * docs/coverage.md
> * docs/cpu.md
> * docs/custom-image.md
> * docs/debug-port.md
> * docs/device_model.md
> * docs/fs.md
> * docs/fuzzing.md
> * docs/gdb.md
> * docs/heap-profiling.md
> * docs/hotplug.md
> * docs/intel_sgx.md
> * docs/intel_tdx.md
> * docs/io_throttling.md
> * docs/iommu.md
> * docs/live_migration.md
> * docs/logging.md
> * docs/macvtap-bridge.md
> * docs/memory.md
> * docs/mshv.md
> * docs/performance_metrics.md
> * docs/profiling.md
> * docs/releases.md
> * docs/seccomp.md
> * docs/snapshot_restore.md
> * docs/tpm.md
> * docs/tracing.md
> * docs/uefi.md
> * docs/vdpa.md
> * docs/vfio-user.md
> * docs/vfio.md
> * docs/vhost-user-blk-testing.md
> * docs/vhost-user-net-testing.md
> * docs/virtiofs-root.md
> * docs/vsock.md
> * docs/windows.md
> * event_monitor/Cargo.toml
> * fuzz/.gitignore
> * fuzz/Cargo.lock
> * fuzz/Cargo.toml
> * hypervisor/Cargo.toml
> * net_gen/Cargo.toml
> * net_util/Cargo.toml
> * option_parser/Cargo.toml
> * pci/Cargo.toml
> * performance-metrics/Cargo.toml
> * rate_limiter/Cargo.toml
> * release-notes.md
> * resources/linux-config-aarch64
> * resources/linux-config-x86_64
> * scripts/check-image-compatibility.sh
> * scripts/common-aarch64.sh
> * scripts/create-cloud-init.sh
> * scripts/gitlint/rules.py
> * scripts/prepare_vdpa.sh
> * scripts/run_integration_tests_aarch64.sh
> * scripts/run_integration_tests_live_migration.sh
> * scripts/run_integration_tests_rate_limiter.sh
> * scripts/run_integration_tests_sgx.sh
> * scripts/run_integration_tests_vfio.sh
> * scripts/run_integration_tests_windows_aarch64.sh
> * scripts/run_integration_tests_windows_x86_64.sh
> * scripts/run_integration_tests_x86_64.sh
> * scripts/run_metrics.sh
> * scripts/run_openapi_tests.sh
> * scripts/run_unit_tests.sh
> * scripts/sha1sums-aarch64
> * scripts/sha1sums-x86_64
> * scripts/test-util.sh
> * scripts/windows-11-iot-enterprise-aarch64.raw.sha1
> * serial_buffer/Cargo.toml
> * test_data/cloud-init/ubuntu/ci/meta-data
> * test_data/cloud-init/ubuntu/ci/network-config
> * test_data/cloud-init/ubuntu/ci/user-data
> * test_data/cloud-init/ubuntu/local/meta-data
> * test_data/cloud-init/ubuntu/local/network-config
> * test_data/cloud-init/ubuntu/local/user-data
> * test_infra/Cargo.toml
> * tpm/Cargo.toml
> * tracer/Cargo.toml
> * vhost_user_block/Cargo.toml
> * vhost_user_net/Cargo.toml
> * virtio-devices/Cargo.toml
> * vm-allocator/Cargo.toml
> * vm-device/Cargo.toml
> * vm-migration/Cargo.toml
> * vm-virtio/Cargo.toml
> * vmm/Cargo.toml
> * vmm/src/api/openapi/cloud-hypervisor.yaml
> ```

Cool - documentation is not Apache-2.0 - see docs/COPYING.md.

Does the reuse config file support comments so we have a have a short explanation - the reason I asked is because to me it looked like this could mean that reuse was only applying to those files.

Reply:
I've added CC-BY-4.0 license for documentation.

reuse [spec](https://reuse.software/spec/) states that
"Each License File MUST be placed in the LICENSES/ directory in the root of the Project. The name of the License File MUST be the SPDX License Identifier of the license followed by an appropriate file extension"

Should I remove old license files as part of this PR?
```
docs/COPYING.docs
LICENSE-BSD-3-Clause
LICENSE-APACHE
```

to download all licenses:
`reuse download --all`

Comments in dep5 comments start with hash mark #
For example: rust-lang [dep5](https://github.com/rust-lang/rust/blob/master/.reuse/dep5)

Reply:
Yes - you can git mv the licenses please.

Reply:
git mv of the licenses is done